specs/PushNotifications/PushNotifications-spec.md

```diff
@@ -0,0 +1,378 @@
+
+# Description
+
+At a high level, we need to provide a way for all Win32 applications to subscribe to and receive Push notifications irrespective of their app type. This includes unpackaged apps and packaged win32 (MSIX, Sparse Signed Packages). Moreover, all Push scenarios should adher to OS resource management policies like Power Saver, Network Attribution, Enterprise group policies etc. The Reunion Push component will abstract away the complexities of dealing with WNS channel registrations and Push related activations as much as possible freeing the developer to focus on other app related challenges. 
+
+We will priorirtize the following feature set for Reunion:
+* Supporting raw push for Packaged Win32(MSIX)
+* Supporting raw push for Unpackaged Win32.
+* Supporting Visual Cloud Toasts for packaged Win32 apps.
+* Supporting Visual Cloud Toasts for unpackaged Win32 apps.
+
+A Portal Registration flow through [AAD (Azure Active Directory)](https://docs.microsoft.com/en-us/azure/active-directory/develop/quickstart-register-app) will also be defined that decouples the dependence of the Push Flow with the Partner Centre Portal. 
+The following are the main steps in the Push flow with AAD:
+* The developer will first obtain a Remote Identifer GUID and a client secret as part of this Portal Registration flow. This is a one time operation by the developer to provision an app. 
+* When requesting a channel using WNS client Reunion APIs, the developer will pass in a remoteId and WNS will return a specially encoded Channel URL with that Remote Identifier. 
+* When the application service Pushes a notification down to the client, it will retrieve an access token with AAD using the client secret provisioned above. 
+* The application service will then pass that access token along with the payload in a Http POST operation to the channel URL. 
+* WNS will ensure that the payload is ultimately delivered to the app residing on the client device.
+
+Phase 1 (Reunion Preview):
+WNS Push APIs will only support WIN32 Packaged app (MSIX).
+
+Phase 2 (Reunion V1):
+Our Push APIs will add support for Unpackaged Win32 scenarios.
+
+# Examples
+
+## Registration of the Push Activator for Win32 (Inproc)
+If a Win32 app needs to subscribe to Push triggers inproc, the code in Main would follow the pattern below:
+* The Registration will take in a CLSID of the COM server as part of the Activator setup. The registration API will simply be a thin wrapper around the BackgroundInfra WinRT APIs and abstract away the COM details from the developer.
+* The app will query ApplifeCycle APIs to retrieve an ActivationKind. (Note: This is covered in a seperate AppLifeCycle API spec.)
```

Comment:
@andreww-msft FYI. Will need a link to your spec to show how we are addressing the new Activation Kinds as discussed in yesterday's meetup. Thanks